File: Creational/Singleton/Python/logger.py
# logger.py

import threading

class Logger:
  _instance = None
  _lock = threading.Lock()  # Ensures thread safety

  def __new__(cls):
    with cls._lock:  # Thread-safe singleton
      if cls._instance is None:
        cls._instance = super(Logger, cls).__new__(cls)
        print("Logger instance created.")
      return cls._instance

  def log(self, message: str):
    print(f"[LOG]: {message}")

File: Creational/Singleton/CSharp/Logger.cs
// Logger.cs

using System;

public sealed class Logger {
  private static Logger? instance;
  private static readonly object lockObject = new();

  private Logger() {
    Console.WriteLine("Logger instance created.");
  }

  public static Logger Instance {
    get {
      lock (lockObject) {
        if (instance == null) {
          instance = new Logger();
        }
        return instance;
      }
    }
  }

  public void Log(string message) {
    Console.WriteLine($"[LOG]: {message}");
  }
}
